kubetest/azure.go

```diff
@@ -1056,20 +1064,54 @@ func (c *Cluster) DumpClusterLogs(localPath, gcsPath string) error {
-	const logDumpURLPrefix string = "https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/hack/log-dump/"
-	logDumpScript, err := downloadFromURL(logDumpURLPrefix+"log-dump.sh", path.Join(c.outputDir, "log-dump.sh"), 2)
-	if err != nil {
-		return fmt.Errorf("error downloading log dump script: %v", err)
-	}
-	if err := control.FinishRunning(exec.Command("chmod", "+x", logDumpScript)); err != nil {
-		return fmt.Errorf("error changing access permission for %s: %v", logDumpScript, err)
+	logDumper := func() error {
+		// Extract log dump script and manifest from cloud-provider-azure repo
+		const logDumpURLPrefix string = "https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/hack/log-dump/"
+		logDumpScript, err := downloadFromURL(logDumpURLPrefix+"log-dump.sh", path.Join(c.outputDir, "log-dump.sh"), 2)
+		if err != nil {
+			return fmt.Errorf("error downloading log dump script: %v", err)
+		}
+		if err := control.FinishRunning(exec.Command("chmod", "+x", logDumpScript)); err != nil {
+			return fmt.Errorf("error changing access permission for %s: %v", logDumpScript, err)
+		}
+		if _, err := downloadFromURL(logDumpURLPrefix+"log-dump-daemonset.yaml", path.Join(c.outputDir, "log-dump-daemonset.yaml"), 2); err != nil {
+			return fmt.Errorf("error downloading log dump manifest: %v", err)
+		}
+
+		if err := control.FinishRunning(exec.Command("bash", "-c", logDumpScript)); err != nil {
+			return fmt.Errorf("error running log collection script %s: %v", logDumpScript, err)
+		}
+		return nil
 	}
-	if _, err := downloadFromURL(logDumpURLPrefix+"log-dump-daemonset.yaml", path.Join(c.outputDir, "log-dump-daemonset.yaml"), 2); err != nil {
-		return fmt.Errorf("error downloading log dump manifest: %v", err)
+
+	logDumperWindows := func() error {
+		const winLogDumpScriptUrl string = "https://raw.githubusercontent.com/adelina-t/windows-testing/logs_collector/scripts/win-ci-logs-collector.sh"
+		winLogDumpScript, err := downloadFromURL(winLogDumpScriptUrl, path.Join(c.outputDir, "win-ci-logs-collector.sh"), 2)
+
+		masterFQDN := fmt.Sprintf("%s.%s.cloudapp.azure.com", c.dnsPrefix, c.location)
+		if err != nil {
+			return fmt.Errorf("error downloading windows logs dump script: %v", err)
+		}
+		if err := control.FinishRunning(exec.Command("chmod", "+x", winLogDumpScript)); err != nil {
+			return fmt.Errorf("error changing permission for script %s: %v", winLogDumpScript, err)
+		}
+		if err := control.FinishRunning(exec.Command("bash", "-c", fmt.Sprintf("%s %s %s %s", winLogDumpScript, masterFQDN, c.outputDir, c.sshPrivateKeyPath))); err != nil {
+			return fmt.Errorf("error while running Windows log collector script: %v", err)
+		}
+		return nil
 	}
 
-	return control.FinishRunning(exec.Command("bash", "-c", logDumpScript))
+	var errors []string
+	if err := logDumper(); err != nil {
+		errors = append(errors, err.Error())
+	}
+	if err := logDumperWindows(); err != nil {
+		errors = append(errors, err.Error())
```

Comment:
would this report errors if there're no Windows nodes in the cluster?

Reply:
No. The collection script will just exit and log that there are no windows nodes it can access.

Reply:
thanks for confirm